src/gamestate.ts

```diff
@@ -1,7 +1,59 @@
+import { areCoordinatesValid } from "./user";
+import { Coordinates } from "./utils";
+
 export interface GameState {
   board: Array<Array<Cell>>;
 }
 
 export interface Cell {
   value: "empty" | "black";
 }
+
+export function generateNewBoard(): GameState {
+  return {
+    board: [
+      [
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+      ],
+      [
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+      ],
+      [
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+      ],
+      [
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+        { value: "empty" },
+      ],
+    ],
+  };
+}
+
+export function updateGameState(
+  previousState: GameState,
+  actionCoordinates: Coordinates
+): GameState {
```

Comment:
Imo `nextMove` would be a better name for `actionCoordinates`, what do you think?